lcls_tools/common/measurements/wire_scan.py

```diff
@@ -214,33 +299,40 @@ def get_profile_range_indices(self):
         # Hold sequential indices (avoid catching return wires)
         profile_idxs = {}
 
-        active_profiles = []
-        if self.my_wire.use_x_wire:
-            active_profiles.append("x")
-        if self.my_wire.use_y_wire:
-            active_profiles.append("y")
-        if self.my_wire.use_u_wire:
-            active_profiles.append("u")
+        ap = self.active_profiles()
 
-        for profile in active_profiles:
+        for p in ap:
             # Get range methods e.g. x_range()
-            method_name = f"{profile}_range"
-            ranges[profile] = getattr(self.my_wire, method_name)
+            method_name = f"{p}_range"
+            ranges[p] = getattr(self.my_wire, method_name)
 
             # Get indices of when position is within a range
             idx = np.where(
-                (position_data >= ranges[profile][0])
-                & (position_data <= ranges[profile][1])
+                (position_data >= ranges[p][0]) & (position_data <= ranges[p][1])
             )[0]
 
+            # Data slice representing a given profile measurement
             pos = position_data[idx]
+
+            # Boolean mask of monotonically non-decreasing data points
+            # Mask of values where difference between neighbors is positive or zero
             mono_mask = np.diff(pos) >= 0
             mono_mask = np.concatenate(([True], mono_mask))
```

Comment:
In the following case you can get monotonically decreasing data:
```
>>> a = np.array([10, 8, 9, 7, 8, 6, 7]);
>>> i = np.concatenate(([True], np.diff(a) >= 0))
>>> a[i]
array([10,  9,  8,  7])
```
If this case is acceptable for the wire scanner then maybe the comment can be reworded. Otherwise maybe another solution is necessary?

Reply:
Oh excellent catch.  I've been struggling to find a decent solution for this for a while.  I haven't got caught with this particular case yet.
I could include something like:

```suggestion
            mono_mask = np.diff(pos) >= 0
            mono_mask = np.concatenate(([True], mono_mask))
            
            # Find first false
            bad = np.where(~mono_mask)
            if bad.size:
                end = bad[0]                # stop before the first decrease
                mono_idx = idx[:end] 
            else:
                mono_idx = idx            # indices were monotonically non-decreasing to start with
```

Reply:
This looks like a really good solution! Thanks, Eloise.  I've implemented this in the code slightly differently, to preserve `mono_mask` and use that to isolate the profile indices.